UMLEditor/UMLEditor/Views/MainWindow.axaml.cs

```diff
@@ -994,6 +916,11 @@ public void ReconsiderCanvasSize()
 
         }
 
+        /// <summary>
+        /// Event handler for view/edit toggle
+        /// </summary>
+        /// <param name="sender">Object that generated the event</param>
+        /// <param name="e">Extra arguments sent to the handler</param>
         private void ViewEditToggle_OnClick(object sender, RoutedEventArgs e)
         {
 
```

Comment:
RenderLines is still referring to classes by the .Name property. The lines are still messed up if you move the box around.

_Notes for RenderLines_:
Having to scan over the canvas' children on each redraw is _very_ expensive and severely lags the editor. At this point we don't need to do too much with making the redraw smarter, we just have to make a few tweaks

- Use GetClassBoxByName to find the source and dest ClassBoxes in RenderLines
- Change GetClassBoxByName to immediately return when we find a matching classbox. We don't need to scan later into the list if we have already found the box. It can return null if a valid box is not found. Don't make it assign and then break out of the loop. We don't want to spend extra time constructing a new ClassBox on a function that will be called very frequently (do away with the "foundClass" local in that method), especially in situations where it is guaranteed that the class exists.

In the future I might do away with the DrawnClassBoxes property because it isn't as smart of an idea as I had thought. We can discuss further optimizations later, such as storing the list of drawn classes in a different data structure. These changes should suffice for now.

I might move the management/ bookkeeping of the rendered classes out to another class in the future.